includes/admin/class-sensei-course-pre-publish-panel.php

```diff
@@ -77,6 +79,33 @@ public function maybe_publish_lessons( $course_id ) {
 			return;
 		}
 
+		$publishing_meta_key = '_sensei_course_publishing_started';
+
+		// Even if the course is already published, each subsequent updates also triggers this hook anyway
+		// which caused the bug https://github.com/Automattic/sensei/issues/7555.
+		// So we need to check if it's an actual publish call.
+		$is_main_publish_call = 'publish' !== $old_status;
+
+		if ( $is_main_publish_call ) {
+			// This is the first call made, it's not the structure saving call, so the added/updated lessons are not yet saved at this point.
+			// So we set the flag to publish lessons on the next call, which is made after the structure is saved.
```

Comment:
I find this comment a bit confusing. You reference a "next call" here, saying that this next call will be made after the structure is saved. But in the previous sentence, you said that this call is also not the structure saving call, so it's unclear from these comments when the call to save the structure actually happens.

Reply:
The saving process is a bit complex :p So I was struggling a bit a capture it in small comments. Basically, when we click on Publish, GB makes a call with the markup. We are referring to it as the _Publish_ call, this is the first call.

If you check the code here -https://github.com/Automattic/sensei/blob/0132d5c48c67abb440bab859fd83d7c4654d75df/assets/shared/structure/structure-store.js#L253, you'll see that we wait for the Save/Publish call to complete before sending the call to save the structure -
https://github.com/Automattic/sensei/blob/0132d5c48c67abb440bab859fd83d7c4654d75df/assets/shared/structure/structure-store.js#L274

After the structure is also saved, we trigger another GB call (not a structure saving call, a same call as the first Publish call, but the difference is by this time, the Course is already published)
https://github.com/Automattic/sensei/blob/0132d5c48c67abb440bab859fd83d7c4654d75df/assets/shared/structure/structure-store.js#L158

And this last call is the one I am referring to in the comments

Reply:
I understand the logic, but I still think the comment needs to be improved a bit to eliminate confusion for the next person. 🙂  Perhaps you could add a numbered list of the calls that are made to the comment? I think it's more important for the comment to be clear than to be succinct.

Reply:
Thanks @donnapep ! I've rewrote all the comments here https://github.com/Automattic/sensei/pull/7582/commits/2ac96e8a96805b8b7c3a4cd46db0f518e4569c06 . Please LMK if you think the new structure is okay. Initally I was trying to use fewer words as I've not seen this many comments in the repo before for a function, so I was afraid if I was overdoing it :p Tried to make it clearer now